### ISSUE TITLE
String was not recognized as a valid DateTime.

### ISSUE THREAD

**Issue by tylerapplebaum** [ryancbutler/Citrix#11]
Hello,

I get this error when running the get-nslicexp.ps1 script. This happens when connecting via HTTP or HTTPS.

```
`PS C:\users\applebaumt\Desktop> .\Get-NSLicense.ps1 -nsip esvpx2 -adminpassword <removed> -https
Connecting HTTPS
Exception calling "ParseExact" with "3" argument(s): "String was not recognized as a valid DateTime."
At C:\users\applebaumt\Desktop\Get-NSLicense.ps1:215 char:1
+ $nsdate = [DateTime]::ParseExact($currentdatestr,"ddd MMM  d HH:mm:ss ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : FormatException

Current NS Time:`
```

This is the netscaler version:

> sh ver
        NetScaler NS10.1: Build 123.1100.e.nc, Date: Feb 27 2014, 08:52:37

This is my PowerShell version:

PS C:\users\applebaumt\Desktop> $PSVersionTable.PSVersion

Major  Minor  Build  Revision
-----  -----  -----  --------
5      0      10586  117

**Comment by ryancbutler** [comment]
Thanks!  Looks like an issue when the day is 1 number vs 2.  December 1 vs December 22

**Comment by ryancbutler** [comment]
Should be fixed.  Had to adjust how the date was looked at with double digit days.  Please try #12 when you get a chance.

**Comment by tylerapplebaum** [comment]
```
Exception calling "ParseExact" with "3" argument(s): "String was not recognized as a valid DateTime."
At C:\users\applebaumt\Desktop\Get-NSLicense.ps1:217 char:1
+ $nsdate = [DateTime]::ParseExact($currentdatestr,"ddd MMM dd HH:mm:ss ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : FormatException

Current NS Time:
```

Same exact error this time.

**Comment by ryancbutler** [comment]
Can you add the line above line 217 and run again?  Need to know what your date string is coming across as.  Tried with a bunch of different dates on my end.

```write-host $currentdatestr```

Should look like.

```#Fix for dates with single day integer
$currentdatestr = $currentdatestr -replace "  "," 0"
write-host $currentdatestr
$nsdate = [DateTime]::ParseExact($currentdatestr,"ddd MMM dd HH:mm:ss yyyy",$null)```

**Comment by tylerapplebaum** [comment]
That variable is null. I'll keep debugging it and see what I can come up with.

**Comment by tylerapplebaum** [comment]
Ah, part of the problem:

http://'my ns IP'/nitro/v1/config/systemfile?args=filelocation:%2Fnsconfig%2Flicense

{ "errorcode": 1232, "message": "Invalid object name [systemfile]", "severity": "ERROR" }

**Comment by ryancbutler** [comment]
Ahhhh.  Looks like file operations weren't added until 10.5.

**Comment by ryancbutler** [comment]
Not much I can do here since file operations aren't available until 10.5.  Added some error handling and a Netscaler function to throw error if anything lower than 10.5
